afqinsight/match.py

```diff
@@ -0,0 +1,139 @@
+"""Perform statistical matching comparing test and control units."""
+
+from os import name
+import numpy as np
+import pandas as pd
+from scipy.optimize import linear_sum_assignment
+
+
+__all__ = ["mahalonobis_dist_match"]
+
+
+def mahalonobis_dist_match(
+    data=None, test=None, ctrl=None, status_col=None, feature_cols=None, threshold=0.2
+):
+    """
+    If data is None, then test and ctrl must be arrays.
+    If data is provided, then status_col and feature_cols are used.
+
+    Parameters
+    ----------
+    data : pandas Dataframe, optional
+        The feature samples for test subjects.
+        Default: None
+    test : array-like of shape (n_samples1, n_features), optional
+        The feature samples for test subjects.
+        Default: None
+    ctrl : array-like of shape (n_samples2, n_features), optional
+        Feature samples for control subjects from which
+        matches will be selected.
+        Default: None
+    status_col : str, optional
+        Column with subject status. Must have only two unique values,
+        evaluating to True and False, where True indicates a test subject
+        and False indicates a control.
+        Default: None
+    feature_cols : list of str, optional
+        List of columns to use as features for MDM.
+        If feature_columns is None then we use all of the columns
+        except status_col as the features.
+        Default: None
+    threshold : float, optional
+        Mahalnobis distance beyond which matches are not accepted.
+        Default: None
+
+    Returns
+    -------
+    
+    """
+    if data is not None:
+        if status_col is None:
+            raise ValueError(
+                ("If data is provided," " status_col must also be provided")
+            )
+
+        if len(data[status_col].unique()) != 2:
+            raise ValueError(
+                ("status_col in data must " "contain only 2 unique values")
+            )
+
+        if feature_cols is None:
+            feature_cols = data.columns
+            feature_cols.remove(status_col)
```

Comment:
I don't think `remove` is a valid method for `pd.Index` instances. Am I missing something?
```suggestion
            feature_cols = data.columns.drop(status_col)
```